File: PythonBase/부르트포스/투포인터/수들의합2.py
import sys
sys.stdin = open('2003.txt', 'r')
input = sys.stdin.readline
N, M = map(int, input().split())
board = list(map(int, input().split()))
result = Sum = left = right = 0
while True:
    if Sum >= M:
        Sum -= board[left]
        left += 1
    elif right == N:
        break
    elif Sum < M:
        Sum += board[right]
        right += 1
    if Sum == M:
        result += 1
print(result)

File: 2021/01/0117js/수들의_합2.js
// var input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")
var input = require("fs")
  .readFileSync("./2003.txt")
  .toString()
  .trim()
  .split("\r\n");

const [N, M] = input[0].split(" ").map(Number);
let nums = input[1].split(" ").map(Number);
let sum = 0,
  left = 0,
  right = 0,
  result = 0;
while (true) {
  if (sum >= M) {
    if (sum === M) result++;
    sum -= nums[left];
    left++;
  } else if (sum < M) {
    if (right === N) break;
    sum += nums[right];
    right++;
  }
}
console.log(result);
